### DISCUSSION TITLE
Geolocating for Caribou

### DISCUSSION THREAD

**Discussion by albertrex** [philipbelesky/Caribou#20] [Ideas]
Caribou is great & convenient. Much more intuitive than doing similar stuff in Elk.
It would be great if it had a geolocating capacity like landarch tools does for nearmap aerials and Elk does for OSM data.

This would meant that using landarchtools + Caribou you could have a scaled, geolocated plan + aerial set out in your file within like 5 min!

Thanks for making Caribou!

P.S. Sorry if this is already there and I've missed it!

**Comment by philipbelesky** [comment]
Hey, thanks for the suggestion. How would the geolocating bit work? Is the idea that you could lookup the lat/lon of a given point in Rhino?

**Reply by albertrex** [reply]
Yes I think that would be the idea! I was hoping that info might be somewhere tucked away in the OSM data file because I thought it was kind of vaguely GIS-adjacent data?

Nearmaps are straightforward I guess because they give you the option to download a .jgw file that's tied to your aerial and locates it using a given map projection  - in Melbourne for cads I think you'd typically  use a GDA (Geocentric Datum of Australia) 2020 projection at MGA (Map Grid of Australia) zone 55.
But then maybe you don't want actual linework PROJECTED (reconciled to earth curvature) in the same way you might an aerial - but I imagine to work it would have to at least use the same zero but I'm afraid I'm not sure where zero is on a GDA projection.

No clue how to go about it myself sorry and only very vague understating of how nearmaps do it but maybe that filetype/ projection info provides some extra contextualising detail?

**Reply by philipbelesky** [reply]
Yea, I think I understand the general problem of trying to reconcile a raster image, a bunch of geometry, and a projection system to all match up and overlap nicely. 

One thing that might be useful is that the `Extract` components from Caribou all have a `Bounds` output which is a rectangle that should match the crop rectangle used when originally downloading the OSM file. Caribou doesn't surface the lat/lon for each of the corners, but you can note them down when cropping or open up the `.osm` file in a text editor and look at the `bounds` values on line 3. Maybe those values can be specified in nearmap (or vice versa?) so that you end up with an image that matches the crop area? The other way to go would be to use one of the corners of the rectangle along with [setearthanchorpoint](https://docs.mcneel.com/rhino/7/help/en-us/commands/earthanchorpoint.htm) to allow rhino to do some of the reconciling.

That all said, Caribou kind of ignores projection systems (see #5). Everything goes through as WGS84 because the math was easy to find and implement. There are a few drop-in code libraries to do other projection systems, but getting those working without sacrificing speed was a little tricky. I should probably just add them in regardless at this point, as it is a pretty severe limitation for going beyond just doing large-scale mapping on the more conceptual end.
